### PR TITLE
First go at doco for the CLI

### PR DESCRIPTION
**Pull request by CEKrause**
Also noted a bug in the config file implementation - made a note in the config README.

### CODE REVIEW

**Comment by CEKrause** [comment]
I'm not sure why my commit history is so long. It should just be the last one. Maybe because I didn't merge in the change to stable before doing the PR? I will fix for future PR (and this one if it makes a difference).

**Comment by MatthewJA** [comment]
Yep, that's why. Let's fix it if we can - otherwise we can squash n merge. Happy to help with that if need be.

**Comment by MatthewJA** [comment]
Tests are failing due to a mismatch between the click parameters and the arguments to the `main` function, so it'd be great if you could fix that!

File: dea_waterbodies/make_time_series.py [hunk 1]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
-@click.option('--end', type=str, default=None)
-@click.option('--size', type=click.Choice(['ALL', 'SMALL', 'HUGE']),
-              default='ALL')
-@click.option('--missing-only/--not-missing-only', default=False)
-@click.option('--skip', type=click.Path(), default=None)
+@click.option('--config', '-c', type=click.Path(), default=None,
+              help='Path to a config.ini file. A config file can be provided '
+              'instead of using the command line options provided.')
+@click.option('--shapefile', type=click.Path(), default=None,
+              help='REQUIRED. Path to the waterbody polygon shapefile (or '
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:173]: --shapefile isn't required if --config is specified.

File: dea_waterbodies/make_time_series.py [hunk 2]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
+              'geojson) you want to run the time series generation for.')
+@click.option('--output', type=click.Path(), default=None,
+              help='REQUIRED. Path to the output directory for the waterbody '
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:176]: Same as for --shapefile

File: dea_waterbodies/make_time_series.py [hunk 3]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
+              'in the --output directory. The default option is to run '
+              'every waterbody polygon in the --shapefile file, and overwrite '
+              'any existing csv files.')
 @click.option('--state', type=click.Choice(
                 ['ACT', 'NSW', 'NT', 'OT', 'QLD', 'SA', 'TAS', 'VIC', 'WA']),
-              default=None)
-@click.option('--no-mask-obs/--mask-obs', default=False)
-@click.option('--all/--some', default=False)
+              default=None,
+              help='This flag allows you to run the analysis for selected '
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:203]: -> "Run analysis for selected states only." Don't need to specify that it's a flag.

File: dea_waterbodies/make_time_series.py [hunk 4]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
+@click.option('--start', type=str, default=None,
+              help='Date string. E.g. 2019-01-01. '
+              'The start date for the waterbody timeseries query. If --start '
+              'is provided --end must also be provided.')
+@click.option('--end', type=str, default=None,
+              help='Date string. E.g. 2019-12-01. '
+              'The end date for the waterbody timeseries query. If --end is '
+              'provided --start must also be provided.')
+@click.option('--missing-only/--not-missing-only', default=False,
+              help='This option specifies whether you want to only run '
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:195]: -> "Whether to only run waterbody polygons that...". Don't need to specify that it's an option.

File: dea_waterbodies/make_time_series.py [hunk 5]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
+              help='REQUIRED. Path to the waterbody polygon shapefile (or '
+              'geojson) you want to run the time series generation for.')
+@click.option('--output', type=click.Path(), default=None,
+              help='REQUIRED. Path to the output directory for the waterbody '
+              'timeseries.')
 @click.option('--time-span', type=click.Choice(['ALL', 'APPEND', 'CUSTOM']),
-              default='ALL')
-@click.option('--output', type=click.Path(), default=None)
+              default='ALL',
+              help='--time-span sets the time range for the waterbody '
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:180]: -> "Sets the time range..." The help will already print out --time-span by itself, so no need to include in the help string.

File: dea_waterbodies/make_time_series.py [hunk 6]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
-              default=None)
-@click.option('--no-mask-obs/--mask-obs', default=False)
-@click.option('--all/--some', default=False)
+              default=None,
+              help='This flag allows you to run the analysis for selected '
+              'Australian states only. This option requires that each polygon '
+              'have an attribute called "STATE" that specifies one of the '
+              'specified state options.')
+@click.option('--no-mask-obs/--mask-obs', default=False,
+              help='This flag allows you to include uncertainties in the '
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:208]: As above

File: dea_waterbodies/make_time_series.py [hunk 7]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
+              'output timeseries. if you specify --mask-obs then you will '
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:209]: --no-mask-obs does this actually

File: dea_waterbodies/make_time_series.py [hunk 8]
```diff
@@ -165,28 +165,63 @@ def get_shapes(config_dict: dict,
...
+              'only filter out timesteps with 100% invalid pixels. You will '
+              'also record the number invalid pixels per timestep.')
```
Comment (MatthewJA) [inline dea_waterbodies/make_time_series.py:211]: One day we should just output this by default. Probably if we get a database

**Review by MatthewJA — CHANGES_REQUESTED** [review]

File: dea_waterbodies/make_time_series.py (hunk 1 above)
Comment (CEKrause) [reply dea_waterbodies/make_time_series.py:173]: Added.

File: dea_waterbodies/make_time_series.py (hunk 2 above)
Comment (CEKrause) [reply dea_waterbodies/make_time_series.py:176]: Added.

File: dea_waterbodies/make_time_series.py (hunk 4 above)
Comment (CEKrause) [reply dea_waterbodies/make_time_series.py:195]: Changed.

File: dea_waterbodies/make_time_series.py (hunk 3 above)
Comment (CEKrause) [reply dea_waterbodies/make_time_series.py:203]: Changed

File: dea_waterbodies/make_time_series.py (hunk 6 above)
Comment (CEKrause) [reply dea_waterbodies/make_time_series.py:208]: Changed.

File: dea_waterbodies/make_time_series.py (hunk 7 above)
Comment (CEKrause) [reply dea_waterbodies/make_time_series.py:209]: Fixed.

**Comment by CEKrause** [comment]
Fixing the commit history. Closing this PR and will make a new one.

**Pull request closed without merging**